File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/ASCIIArt.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class ASCIIArt(val art: String)

suspend fun main() {
  val art: ASCIIArt = OpenAI.conversation { prompt("ASCII art of a cat dancing") }
  println(art)
}

File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/ASCIIArt.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import com.xebia.functional.xef.prompt.Prompt
import io.circe.Decoder

private final case class ASCIIArt(art: String) derives SerialDescriptor, Decoder

@main def runASCIIArt: Unit =
  lazy val asciiArt = conversation {
    prompt[ASCIIArt](Prompt("ASCII art of a cat dancing"))
  }
  println(asciiArt.art)
